### DISCUSSION TITLE
how can i get the element ”canvas“ when I set the link flag PROXY_TO_PTHREAD=1

### DISCUSSION THREAD

**Discussion by WenAce** [emscripten-core/emscripten#19965] [Q&A]
I need to use emscripten_webgl_create_context to create content，when i set the PROXY_TO_PTHREAD=1，emscripten_webgl_create_context can‘t create content，the value is 0，because it can’t get the element ‘canvas’

**Comment by jspanchu** [comment]
I think you're not using the correct compile/link flags.

Refer to [test/gl_in_proxy_pthread.cpp](https://github.com/emscripten-core/emscripten/blob/main/test/gl_in_proxy_pthread.cpp) and it's compiler flags in https://github.com/emscripten-core/emscripten/blob/0be5b609412d5b07f8157528df8e5088dae84858/test/test_browser.py#L4624
